### PR TITLE
Fix child reorder

### PR DESCRIPTION
**Pull request by juliangruber**
It was easier for me to write a new algorithm than to understand what was there. I hope, the new algorithm is clearer :)

This for sure needs the change to on-load, because every child will be removed and re-added.

I will verify this in dat-desktop now.

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
Got two nits:
1. I suspect by always removing all child nodes from the DOM we'll give paints a _very_ bad time. If we can find a way to only remove them for a reorder that'd help a massive amount
2. probably shouldn't check in `package-lock.json`; adding it to `.gitignore` would be good haha

**Comment by juliangruber** [comment]
Forgot to mention: This will now also work when there's nodes with `ID` and without mixed in a container.

**Comment by juliangruber** [comment]
this works in microcomponent's test suite and example script, but not dat-desktop yet

File: test/diff.js [hunk 1]
```diff
@@ -344,6 +345,7 @@ tape('use id as a key hint', function (t) {
     var oldForth = a.children[3]
 
     var c = nanomorph(a, b)
+    console.log('c', c.outerHTML)
```
Comment (yoshuawuyts) [inline test/diff.js]: I'm thinking we should probably strip these before we merge (:

**Comment by yoshuawuyts** [comment]
@juliangruber should we merge this as an intermediate thing? Any way I can help?

**Comment by juliangruber** [comment]
doesn't matter too much to me, either are worky-ish broken

**Comment by yoshuawuyts** [comment]
oh alright, well if they're both broken that's no good :/

**Comment by yoshuawuyts** [comment]
📦 `v5.1.1`

**Comment by juliangruber** [comment]
❤️ ❤️ ❤️ ❤️

**Pull request merged**